src/onehot/problem_description.cpp

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/onehot/problem_description.hpp>
+#include <miopen/names.hpp>
+
+#include <sstream>
+
+namespace miopen {
+
+namespace onehot {
+
+NetworkConfig ProblemDescription::MakeNetworkConfig() const
+{
+    auto dtype       = inDesc.GetType();
+    long outputNumel = inputSize * numClasses;
+
+    std::ostringstream ss;
+
+    ss << "onehot";
+    ss << "dtype" << dtype;
```

Comment:
The string generated here is used to distinguish the result.invoker_factory created in the solver.
If there is the same key, the solver does not run again but reuses the already generated result.invoker_factory.
When looking at the contents of the solver, it seems that AlignUp(problem.getInputSize(), xlocalsize);, input_dtype, and output_dtype are needed.
Since numClasses is passed as an argument to the kernel, it does not need to be distinguished here.

Reply:
I'm sorry. Information about skipping with IsApplicable(), etc. is also needed.
Please consider that the solver will not be called if there is an identical string key.

Reply:
I have updated the cache key as below: